app/controllers/AccountManagement.scala

```diff
@@ -351,22 +354,29 @@ object AccountManagement extends Controller with ContextLogging with CatalogProv
     implicit val tpBackend = resolution.backend
     val eventualMaybeSubscription = SessionSubscription.subscriptionFromRequest
     val errorCodes = errorCode.toSeq.flatMap(_.split(',').map(_.trim)).filterNot(_.isEmpty).toSet
+    val futureMaybeMaybeEmail = (for {
+      authUser <- OptionT(Future.successful(authenticatedUserFor(request)))
+      idUser <- OptionT(IdentityService.userLookupByCredentials(authUser.credentials))
+    } yield Option(idUser.primaryEmailAddress)).run.recover{case _ => None}
+
 
     val futureMaybeFutureManagePage = for {
       subscription <- OptionT(eventualMaybeSubscription).filter(!_.isCancelled)
       account <- OptionT(tpBackend.zuoraService.getAccount(subscription.accountId).map{Some(_)}.recover{case t: Throwable => None})
       pendingHolidays <- OptionT(tpBackend.suspensionService.getUnfinishedHolidays(subscription.name, now).map(_.toOption))
       billingSchedule <- OptionT(tpBackend.commonPaymentService.billingSchedule(subscription.id, account, numberOfBills = 13).map(Some(_):Option[Option[BillingSchedule]]))
+      maybeEmail <- OptionT(futureMaybeMaybeEmail)
```

Comment:
so just to clarify if there's no user in the session or the lookup fails to find anything, you want to quit the for comprehension, however if there a user there but there null returned or an exception is thrown, you want to carry on with None as the email.
I'm wondering vaguely whether there might be a better type to use than Option[Option[...]] just to avoid confusion